pkg/services/mto_service_item/mto_service_item_creator.go

```diff
@@ -342,15 +342,44 @@ func (o *mtoServiceItemCreator) makeExtraSITServiceItem(appCtx appcontext.AppCon
 		}
 	}
 
-	extraServiceItem := models.MTOServiceItem{
-		MTOShipmentID:   firstSIT.MTOShipmentID,
-		MoveTaskOrderID: firstSIT.MoveTaskOrderID,
-		ReServiceID:     reService.ID,
-		ReService:       reService,
-		SITEntryDate:    firstSIT.SITEntryDate,
-		SITPostalCode:   firstSIT.SITPostalCode,
-		Reason:          firstSIT.Reason,
-		Status:          models.MTOServiceItemStatusSubmitted,
+	//When a DDFSIT is created, this is where we auto create the accompanying DDASIT and DDDSIT
+	var extraServiceItem models.MTOServiceItem
```

Comment:
suggestion: is it possible to skip this if/else condition entirely and just copy the CustomerContacts like we copy most of the other fields? 

```suggestion
		extraServiceItem := models.MTOServiceItem{
		MTOShipmentID:   firstSIT.MTOShipmentID,
		MoveTaskOrderID: firstSIT.MoveTaskOrderID,
		ReServiceID:     reService.ID,
		ReService:       reService,
		SITEntryDate:    firstSIT.SITEntryDate,
		SITPostalCode:   firstSIT.SITPostalCode,
		Reason:          firstSIT.Reason,
		Status:          models.MTOServiceItemStatusSubmitted,
		CustomerContacts: firstSIT.CustomerContacts,
	}
```

Reply:
@michaelvaldes-truss My motivation behind the if/else block is two fold. 

- I initially tried out your suggested change as it seemed like the most straight forward approach but when you try to submit a DDFSIT as a Prime user you are met with a 500 error. From what I understand, this is because it is adding unexpected fields when the only fields `CustomerContacts` is expecting is ` Type`, `FirstAvailableDeliveryDate`, and `TimeMilitary`. That's the story behind this block of code:
```
 CustomerContacts: []models.MTOServiceItemCustomerContact{
      {
	     Type:                       models.CustomerContactTypeFirst,
	     FirstAvailableDeliveryDate: firstSIT.CustomerContacts[0].FirstAvailableDeliveryDate,
	     TimeMilitary:               firstSIT.CustomerContacts[0].TimeMilitary,
      },
      {
	     Type:                       models.CustomerContactTypeSecond,
	     FirstAvailableDeliveryDate: firstSIT.CustomerContacts[1].FirstAvailableDeliveryDate,
	     TimeMilitary:               firstSIT.CustomerContacts[1].TimeMilitary,
      },
 },
```

- That resolved our 500 error but if you tried running the command `make db_dev_fresh` you would be met with the below error. so I included the if/else statement to account for that.
```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
```


Open to suggestions if you have a better approach

Reply:
I gotcha! Here's what I suspect is happening:

If you just set the customer contacts directly as in
`CustomerContacts: firstSIT.CustomerContacts,`
the extra service item shares the same exact customer contacts as the original service item, including their db ids. 
When the `mto_service_item_creator` creates each extra service item, it probably tries to explicitly create a new customer contact and runs into a duplicate table id problem.

The way you wrote it actually creates a new customer contact struct without an ID, and so the `mto_service_item_creator` has no problem creating a new customer contact with a new ID and the same Type, FADD, and Time.

If this is the case, then we probably want to ask the following question: should these service items 
a) explicitly share the same customer contact in the database, or 
b) should they have different customer contacts with the same data (at least upon creation)?

If the answer is a) then we should update the `mto_service_item_creator` to not break when saving a customer contact that already exists.
If the answer is b) then we might want to create a helper function like `copyCustomerContact` that does pretty much exactly what you have now, so that we can call it and not have to reference specific indices of the customerContacts array.

Reply:
After conferring with @donaldthai during our 1:1, came to the conclusion the answer is b so I created a helper function to copy the contacts over without relying on indices to keep things flexible 🎉